packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/ObjectBoundRealmResults.kt

```diff
@@ -78,5 +78,8 @@ internal class ObjectBoundRealmResults<E : BaseRealmObject>(
  * Binds a flow to an object lifecycle. It allows flows on queries to complete once the object gets
  * deleted. It is used on sub-queries and backlinks.
  */
-internal fun <T> Flow<T>.bind(reference: RealmObjectReference<out BaseRealmObject>): Flow<T> =
-    this.terminateWhen(reference.asFlow()) { it is DeletedObject }
+internal fun <T> Flow<T>.bind(
+    reference: RealmObjectReference<out BaseRealmObject>,
+    keyPaths: List<String>?
```

Comment:
Do we actually need the `keyPaths` here to register if the object is deleted? It sounds a bit strange if the deletion event is not triggered with out a specific path.

Reply:
Good point. Removing it.